### ISSUE TITLE
Fix EKS NodeGroup Parameter Definitions

### ISSUE THREAD

**Issue by wesleywh** [cloudtools/troposphere#1574]
When defining a NodeGroup (EKS) these two parameters need to be updated:

`Labels` and `Tags`

In `eks.py` it should be modified to fit:
```
class Nodegroup(AWSObject):
    resource_type = "AWS::EKS::Nodegroup"

    props = {
        'AmiType': (basestring, False),
        'ClusterName': (basestring, True),
        'DiskSize': (double, False),
        'ForceUpdateEnabled': (boolean, False),
        'InstanceTypes': ([basestring], False),
        'Labels': (dict, False),
        'NodegroupName': (basestring, False),
        'NodeRole': (basestring, True),
        'ReleaseVersion': (basestring, False),
        'RemoteAccess': (RemoteAccess, False),
        'ScalingConfig': (ScalingConfig, False),
        'Subnets': ([basestring], False),
        'Tags': (dict, False),
        'Version': (basestring, False),
    }
```

The change is:
Labels = basestring -> dict
Tags = Tags -> dict

I was unable to deploy with the previous settings because I kept getting errors from cloudformation. After the change I was able to deploy correctly.

Here is an example of the deploy:
```
template.add_resource(Nodegroup(
        "NodeGroup",
        AmiType=Ref("AmiType"),
        ...
        ...
        Labels=json.loads(sceptre_user_data.get("Labels")) or Ref("AWS::NoValue"),
        Tags={ 
           "MyKey": "MyValue",
           "MySecondKey": "MySecond Value"
        }
))
```

**Comment by PatMyron** [comment]
```
          "PrimitiveType": "Json",
```
for `AWS::EKS::Nodegroup.Labels` and `AWS::EKS::Nodegroup.Tags` in the [Resource Specification](https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json)

---

https://github.com/cloudtools/troposphere/blob/74a59ccafa37c7933379d1e9e3da4206e149a731/troposphere/eks.py#L66-L84

---

`Labels` seems like it needs to change type, but I'm not sure how the `Tags` type works @markpeek

**Comment by markpeek** [comment]
@PatMyron heh...don't get me started about Tags... :-) Here's the thing, the Tags() helper was added given the verbosity of the original tags with Key/Value keys. Lately CloudFormation has seen Tags being of type Json without using those keys. I'm wondering if this has changed recently or was originally missed. For both those types the likely fix is to allow "dict" and the existing type for backwards compatibility. Perhaps something like:
```
diff --git a/troposphere/eks.py b/troposphere/eks.py
index bc64a55..66c7c7a 100644
--- a/troposphere/eks.py
+++ b/troposphere/eks.py
@@ -72,13 +72,13 @@ class Nodegroup(AWSObject):
         'DiskSize': (double, False),
         'ForceUpdateEnabled': (boolean, False),
         'InstanceTypes': ([basestring], False),
-        'Labels': (basestring, False),
+        'Labels': ((basestring, dict), False),
         'NodegroupName': (basestring, False),
         'NodeRole': (basestring, True),
         'ReleaseVersion': (basestring, False),
         'RemoteAccess': (RemoteAccess, False),
         'ScalingConfig': (ScalingConfig, False),
         'Subnets': ([basestring], False),
-        'Tags': (Tags, False),
+        'Tags': ((Tags, dict), False),
         'Version': (basestring, False),
     }
```
Although the Tags likely only work one way.

**Comment by dpoirier** [comment]
I ran into this too. The Tags would need to be serialized as a Json object when they are part of a NodeGroup, and it's not obvious to me how that could be accomplished. If Tags are used in my Troposphere definition of a NodeGroup, I get an array of Tags instead of a single Json object in my template, and stack creation fails.

**Comment by mromaszewicz** [comment]
Both "Labels" and "Tags" should simply be a dict, the alternate `basestring` and `Tags` representations simply don't work in AWS.

I've made that substitution in a local fork, and it works great. I don't think you need to worry about troposphere backward compatibility since the string and Tags representations never worked.

**Comment by markpeek** [comment]
Fixed via #1576
